src/util/stream.rs

```diff
@@ -19,6 +22,35 @@ use std::time::Duration;
 ///
 /// [`timeout`]: #method.timeout
 pub trait StreamExt: Stream {
+    /// Debounce the stream on the trailing edge using the given duration.
+    ///
+    /// Errors will pass through without being debounced. Debouncing will
+    /// happen on the trailing edge. This means all items (except the last
+    /// one) will be discarded until the delay has elapsed without an item
+    /// being passed through. The last item that was passed through will
+    /// be returned.
+    ///
+    /// See also [`debounce_builder`], which allows more configuration over how the
+    /// debouncing is done.
+    ///
+    /// [`debounce_builder`]: #method.debounce_builder
+    fn debounce(self, dur: Duration) -> Debounce<Self>
```

Comment:
I added this specialized combinator because I imagine this is the version how most people are going to use this. 

We may even consider adding another combinator that sets just the `max_wait` in order to [`sample`](http://reactivex.io/documentation/operators/sample.html) the underlying stream at the given interval. Right now, sampling functionality is integrated with `debounc`ing, so the second combinator would be there just for convenience.

Reply:
I just added a `sample` combinator in e7784d3 because I believe the case is common as well. I'll revert the commit if you don't want it in. :)